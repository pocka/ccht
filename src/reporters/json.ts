import type { Reporter, CheckResult } from "../types";

export interface JsonReporterOptions {
  /**
   * Pretty print JSON?
   * @default false
   */
  pretty?: boolean;
}

export class JsonReporter implements Reporter {
  #pretty: boolean;

  constructor({ pretty = false }: JsonReporterOptions) {
    this.#pretty = pretty;
  }

  async report({ links }: CheckResult) {
    return JSON.stringify(links, null, this.#pretty ? 2 : 0);
  }
}
