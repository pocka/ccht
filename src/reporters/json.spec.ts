import { JsonReporter } from "./json";

it("should emit valid JSON", async () => {
  const reporter = new JsonReporter({});

  const json = await reporter.report({
    links: [
      {
        type: "error",
        severity: "danger",
        url: "https://example.com/foo/bar.baz",
        reason: 'Foo\nBar"Baz"',
        referredBy: ["https://example.com"],
      },
    ],
  });

  expect(() => {
    JSON.parse(json);
  }).not.toThrow();
});

it("should match snapshot", async () => {
  const reporter = new JsonReporter({});

  expect(
    await reporter.report({
      links: [
        {
          type: "error",
          severity: "danger",
          url: "https://example.com/foo/bar.baz",
          reason: 'Foo\nBar"Baz"',
          referredBy: ["https://example.com"],
        },
      ],
    })
  ).toMatchSnapshot();
});
