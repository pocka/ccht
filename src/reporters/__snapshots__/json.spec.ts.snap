// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `"[{\\"type\\":\\"error\\",\\"severity\\":\\"danger\\",\\"url\\":\\"https://example.com/foo/bar.baz\\",\\"reason\\":\\"Foo\\\\nBar\\\\\\"Baz\\\\\\"\\",\\"referredBy\\":[\\"https://example.com\\"]}]"`;
